tests/vmi_configuration_test.go

```diff
@@ -2110,4 +2110,31 @@ var _ = Describe("Configurations", func() {
 			Expect(err).ToNot(HaveOccurred())
 		})
 	})
+
+	It("VMI with masquerade binding and guest agent should expose Pod IP as its public address", func() {
+		vmi := tests.NewRandomFedoraVMIWitGuestAgent()
+
+		By("Starting a VirtualMachineInstance")
+		vmi, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+		Expect(err).ToNot(HaveOccurred())
+		tests.WaitForSuccessfulVMIStart(vmi)
+
+		vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmi.Name, &metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		oldMacOnVMI := vmi.Status.Interfaces[0].MAC
+
+		tests.WaitAgentConnected(virtClient, vmi)
+
+		vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+		Eventually(func() string {
+			vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmi.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			return vmi.Status.Interfaces[0].IP
+		}, 180*time.Second, 1*time.Second).Should(Equal(vmiPod.Status.PodIP))
```

Comment:
Can you add a brief explanation as to what was being tested for in case of failure?

`Error: <something> != <something>`

makes devs have to inspect the code to figure out what went wrong.

Reply:
Added